### ISSUE TITLE
Some OData responses need to be wrapped in a special OData type

### ISSUE THREAD

**Issue by travis-leith** [Zaid-Ajaj/Hawaii#23]
Similar to #21, most OData responses come back with a field `odata.context`.

So we need a generic type

```F#
type OData<'a> =
    {
        [<JsonProperty("@odata.context")>]
        ODataContext: string
        value: 'a
    }
```

and instead of generating something like

```F#
[<RequireQualifiedAccess>]
type Extract =
    ///Success
    | OK of payload: SomeType
    ///error
    | DefaultResponse of payload: odataerror
```

the following is needed

```F#
[<RequireQualifiedAccess>]
type Extract =
    ///Success
    | OK of payload: OData<SomeType>
    ///error
    | DefaultResponse of payload: odataerror
```

in cases that return the odata context.

**Comment by Zaid-Ajaj** [comment]
Can you please elaborate? The payloads of the responses are derived from the schema so if `value` is expected there, it should be included already. Is that not the case? As of `odata.context`, if this is only for response payloads and we don't include it in the type, it won't affect deserialization I believe.

If you have a test schema, specific operation and types that I can look into, that would be great 🙏

**Comment by travis-leith** [comment]
schema: "https://selectapi.datascope.refinitiv.com/RestApi/v1/Authentication/$metadata"

snippet of spec:

```
"responses": {
          "200": {
            "description": "Success",
            "content": {
              "application/json": {
                "schema": {
                  "type": "string",
                  "nullable": true
                }
              }
            }
          },
          "default": {
            "$ref": "#/components/responses/error"
          }
        }
```
I think what is happening is that in cases of primitive return values, it wraps the result in the above mentioned type so that it can include the context. I have now also encountered a non primitive return type that des no such wrapping and the context is included as an additional field.

**Comment by Zaid-Ajaj** [comment]
Fixed as of v0.49 🚀 I am following your lead and wrapping _only_ primitive responses (string, bool, numbers, guids etc.) with a `ODataContext<'TValue>` type for deserialization. Please give it a try and let me know if we are doing this right 😄

**Comment by travis-leith** [comment]
Looks good for primitive types, but I think it is also required for lists of anything, since the same principle applies: the response needs to have a place at the top level to indicate the odata.context.

For example, the following comes back wrapped similarly

```
"responses": {
          "200": {
            "description": "Success",
            "content": {
              "application/json": {
                "schema": {
                  "type": "array",
                  "items": {
                    "nullable": true,
                    "$ref": "#/components/schemas/DataScope.Select.Api.Extractions.ExtractionRequests.ExtractionRow"
                  }
                }
              }
            }
          },
          "default": {
            "$ref": "#/components/responses/error"
          }
        }
```

**Comment by Zaid-Ajaj** [comment]
Fixed as of v0.51 now it also wraps arrays of things 😄
